WebApp/autoreduce_webapp/templates/instrument_summary.html

```diff
@@ -1,48 +1,175 @@
+                                </div>
+                            {% endfor %}
+                        </div>
+                    {% elif filtering == 'experiment' %}
+                    <!-- Experiment filtered jobs -->
+                        <div class="row" id="by-experiment-number">
+                            {% for experiment, associated_runs in experiments.items %}
+                            {% if forloop.first %}
+                                <div class="col-md-12 run-row-top" data-toggle="collapse" data-target="#RB{{ experiment.reference_number }}">
+                            {% else %}
+                                <div class="col-md-12 run-row" data-toggle="collapse" data-target="#RB{{ experiment.reference_number }}">
+                            {% endif %}
+                                <div class="col-md-1"></div>
+                                <a href="{% url 'experiment_summary' experiment.reference_number %}"> RB{{ experiment.reference_number }}</a>
+                            </div>
+                            <div id="RB{{ experiment.reference_number }}" class="collapse">
+                                {% for run in associated_runs %}
+                                <div class="row run-row-internal">
+                                    <div class="col-md-5 col-sm-5 col-md-offset-1 col-xs-4"><a href="{% url 'run_summary' instrument_name=instrument_name run_number=run.run_number run_version=run.run_version %}">{{ run.title }}</a></div>
+                                    <div class="col-md-2 col-sm-2 col-xs-4 run-status text-{% colour_table_row run.status.value %}"><strong>{{ run.status }}</strong></div>
+                                    <div class="col-md-4 col-sm-5 col-xs-4"><strong>Last updated:</strong> {{ run.last_updated }}</div>
+                                </div>
+                                {% endfor %}
+                            </div>
+                            {% endfor %}
+                        </div>
+                    {% else %}
+                        <div class="row">No runs to show</div>
+                    {% endif %}
+                </div>
+            </div>
+            <br/>
+            <!-- Paginator -->
+            {% if filtering == 'run' %}
+                <div class="row">
+                    {% if paginator %}
+                    <div class="col-md-12 text-center">
+                        <div class="btn-group" role="group">
+                            {% if paginator.has_previous %}
+                                <button title="First Page" class="btn btn-default" onclick="changePage(1)"><i class="fa fa-step-backward"></i></button>
+                                <button title="Previous Page" class="btn btn-default" onclick="changePage({{ paginator.previous_page_index }})"><i class="fa fa-chevron-left"></i></button>
+                            {% endif %}
+                            {% for page in paginator.display_list %}
```

Comment:
It seems like some of this processing could be removed from the template. A method called get_visible_page could be implemented or similar.

Reply:
Which code are you thinking can be removed? I'm not sure that I completely follow.

Given that the display list is length of 1 + page_tolerance * 2 where page tolerance defaults to 3, it should never be processing more than a list of length 9. 

I think it could potentially reduce readability more than it's worth to performance  by changing this - unless I misunderstand what you are considering changing?